src/promo/promo.repository.ts

```diff
@@ -0,0 +1,68 @@
+import { plainToInstance } from 'class-transformer';
+
+import { appDataSource } from '../../data-source';
+import { PromocodeEntity } from './entity/promo.entity';
+import { IAddPromoCode, IPageOptions, IPromoCode, IPromoCodeName, IRemovePromoCode } from './promo.interface';
+import { PromocodeState } from './promocode.enum';
+
+export class PromocodeRepository {
+  async getListOfPromocodes(pageOptions: IPageOptions): Promise<PromocodeEntity[]> {
+    let query = 'SELECT * FROM promocode';
+
+    if (!pageOptions.page) {
```

Comment:
in promo.proto this field is required. the same about limit. do we need this check?